src/package.cc

```diff
@@ -52,9 +68,34 @@ std::pair<double, int> Package::GetFillMass(double qty) {
       fill_mass = fill_max_;
     }
   }
-  fill_mass = std::min(qty, fill_mass);
-  num_at_fill_mass = static_cast<int>(std::floor(qty / fill_mass));
-  return std::pair<double, int>(fill_mass, num_at_fill_mass);
+
+  if (strategy_ == "first" || strategy_ == "equal") {
+    fill_mass = std::min(qty, fill_mass);
+    num_at_fill_mass = static_cast<int>(std::floor(qty / fill_mass));
+    ExceedsSplitLimits(num_at_fill_mass);
+    packages.assign(num_at_fill_mass, fill_mass);
+
+    qty -= num_at_fill_mass * fill_mass;
+  }
+
+  if (strategy_ == "uniform" || strategy_ == "normal") {
+    // only use random if a full package amount is available. if less than one
+    // full amount is available, below will fill a partial package (no random).
+    while ((qty > 0) && (qty >= fill_max_)) {
+      fill_mass = dist_->sample();
+      packages.push_back(fill_mass);
+      qty -= fill_mass;
+    }
+  }
+
+  if (qty > eps_rsrc() && qty >= fill_min_) {
```

Comment:
I think I know why these are not redundant because it's not inconceivable for `fill_min_ < eps_rscr()`, but we could again do:
```suggestion
  if (qty > std::max(eps_rsrc(),fill_min_) ) {
```